storage/database/db_manager.go

```diff
@@ -882,6 +886,41 @@ func (dbm *databaseManager) TryCatchUpWithPrimary() error {
 	return nil
 }
 
+func (dbm *databaseManager) Stat(property string) (string, error) {
```

Comment:
I think the suggestion may replace only the following statement.
 `headInfo := fmt.Sprintf(" [%s:%s]\n", DBEntryType(idx), db.Type())` 
Also, rather the collecting shard database header(string) together once, getting one by one seems efficient in the stat function, if not, it adds code to modify them.

I think this version seems succinct. If I'm missing your point, please let me know again. Thanks.

Reply:
I was unclear about the intention.

1. Returning a JSON-able structure is convenient in the downstream (i.e. in RPC clients and node console)
```js
> debug.chaindbProperty("leveldb.stats")
[
  { dbEntry: "body", dbType: "LevelDB", value: "abcd" },
  { dbEntry: "state", dbType: "LevelDB", value: "0000" }, // sharded
  { dbEntry: "state", dbType: "LevelDB", value: "1111" },
  { dbEntry: "state", dbType: "LevelDB", value: "2222" },
  { dbEntry: "state", dbType: "LevelDB", value: "3333" },
]
> debug.chaindbProperty("leveldb.stats").filter((prop) => prop.dbEntry == "body")[0].value
"abcd"
```

2. The `type Database interface` already has the `GetProperty(name)` method. Instead of adding `Stat`, we can modify the existing `GetProperty` instead. `GetProperty` is only used in PrivateDebugAPI anyway.

3. For the shard database, we can modify the method type `GetProperty(name) ([]string, error)`

```go
type Database interface {
  GetProperty(name) ([]string, error)
}

func (db *levelDB) GetProperty(name string) ([]string, error) {
  value, err := db.db.GetProperty(name)
  return []string{value}, err
}

func (db *shardedDB) GetProperty(name string) ([]string, error) {
  var results []string
  var errs []error
  for idx, shard := range db.shards {
    values, err := shard.Property(name)
    results = append(results, values...)
    errs = append(errs, err)
  }
  return results, errors.Join(errs)
}

type DBProperty struct {
  Entry DBEntryType `json:"dbEntry"`
  Type DBType `json:"dbType"`
  Value string `json:"value"`
}

type DBManager interface {
  GetProperty(name) ([]DBProperty, error)
}

func (dbm *databaseManager) GetProperty(name string) ([]DBProperty, error) {
  var results []string
  var errs []error
  for idx, db := range dbm.dbs {
    if db != nil {
      values, err := db.GetProperty(name)
      for _, value := range values {
        results = append(results, DBProperty{
          Entry: DBEntryType(idx),
          Type: db.Type(),
          Value
        }
      }
      errs = append(errs, err)
    }
  }
  return results, errors.Join(errs)
}
```

Reply:
On the second tought, the property value is quite large so it cannot fit into one string.
Forget about my comments.